mlir/lib/Dialect/Func/Transforms/FuncConversions.cpp

```diff
@@ -37,7 +37,8 @@ struct CallOpSignatureConversion : public OpConversionPattern<CallOp> {
     // Substitute with the new result types from the corresponding FuncType
     // conversion.
     rewriter.replaceOpWithNewOp<CallOp>(
-        callOp, callOp.getCallee(), convertedResults, adaptor.getOperands());
+        callOp, callOp.getCallee(), convertedResults,
+        getOneToOneAdaptorOperands(adaptor.getOperands()));
```

Comment:
Aha, I see, thanks. The diff view in Github had hidden that check... Another question though: is the test really enough? Can't we have a conversion `a -> (c, d)` of one operand and `b -> ()` for another, such that the operands `a, b` get converted to `c, d`, which passes that test but then makes `getOneToOneAdaptorOperands` fail? (Since we want to ensure that there are really only 1:1 conversions, using that function is actually a good idea, I think.)

Reply:
Good point. We should be using `rewriter.replaceOpWithMultiple` to support 1:N replacements for the results. I put this into a new PR so that this PR does not get any larger. See #117413. Will rebase this PR on top of #117413.

Also, you're right about `getOneToOneAdaptorOperands`. We should be calling `flattenOperands`. It's in the callee where the N:1 mapping must be recovered, not all the call site. I was confusing op results and operands.